zyte_common_items/pages.py

```diff
@@ -1,35 +1,97 @@
 from datetime import datetime
+from typing import Generic, Type, TypeVar
 
 import attrs
 from web_poet import ItemPage, RequestUrl, Returns, WebPage, field
 from web_poet.pages import ItemT
 
-from .components import Metadata
-from .items import Article, ArticleList, BusinessPlace, Product, ProductList, RealEstate
-from .util import format_datetime
+from .components import (
+    ArticleListMetadata,
+    ArticleMetadata,
+    BusinessPlaceMetadata,
+    ProductListMetadata,
+    ProductMetadata,
+    ProductNavigationMetadata,
+    RealEstateMetadata,
+    request_list_processor,
+)
+from .items import (
+    Article,
+    ArticleList,
+    BusinessPlace,
+    Product,
+    ProductList,
+    ProductNavigation,
+    RealEstate,
+)
+from .util import format_datetime, metadata_processor
+
+try:
+    from typing import get_args as _get_args
+except ImportError:
+
```

Comment:
could you please add a comment mentioning Python version, so that we don't forget to remove the workaround later?

Reply:
It's added in 3.8, I think we should just drop 3.7.